stock_location_lockdown/models/stock_quant.py

```diff
@@ -9,18 +9,13 @@
 class StockQuant(models.Model):
     _inherit = 'stock.quant'
 
-    @api.model
-    def quants_move(
-            self, quants, move, location_to, location_from=False, lot_id=False,
-            owner_id=False, src_package_id=False, dest_package_id=False,
-            entire_pack=False):
-        if (
-                location_to.usage == 'internal' and
-                location_to.block_stock_entrance):
-            raise UserError(_(
-                "The location '%s' is not configured to receive stock.")
-                % location_to.display_name)
-        return super(StockQuant, self).quants_move(
-            quants, move, location_to, location_from=location_from,
-            lot_id=lot_id, owner_id=owner_id, src_package_id=src_package_id,
-            dest_package_id=dest_package_id, entire_pack=entire_pack)
+    @api.constrains('location_id')
+    def _check_location_blocked(self):
+        for record in self:
+            if record.location_id.block_stock_entrance and record.location_id.usage == 'internal':
```

Comment:
The second condition seems useless. If `block_stock_entrance` is checked, we do not want quant anyway, whatever the usage, no?

Reply:
your right